server/blob/blob-sharing-file/src/main/java/org/apache/james/blob/sharing/file/SimpleBlobSharingMechanism.java

```diff
@@ -0,0 +1,120 @@
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.blob.sharing.file;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.james.blob.api.BlobId;
+import org.apache.james.blob.api.BlobStore;
+import org.apache.james.blob.sharing.api.BlobSharingMechanism;
+import org.apache.james.core.MailAddress;
+import org.apache.james.core.builder.MimeMessageBuilder;
+import org.apache.james.filesystem.api.FileSystem;
+import org.apache.james.server.core.MailImpl;
+import org.apache.mailet.MailetContext;
+
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+public class SimpleBlobSharingMechanism implements BlobSharingMechanism {
+    private static final int STRING_LENGTH = 32;
+    private static final boolean WITH_LETTERS = true;
+    private static final boolean WITH_NUMBERS = true;
+    private static final String SUBJECT = "A blob had been shared with you";
+    static final String CORRESPONDING_FILE_HEADER = "corresponding-file";
+
+    public static class Configuration {
+        private final String sharingDirectory;
+
+        public Configuration(String sharingDirectory) {
+            this.sharingDirectory = sharingDirectory;
+        }
+    }
+
+    private final MailetContext mailetContext;
+    private final BlobStore blobStore;
+    private final FileSystem fileSystem;
+    private final Configuration configuration;
+
+    SimpleBlobSharingMechanism(MailetContext mailetContext, BlobStore blobStore, FileSystem fileSystem, Configuration configuration) {
+        this.mailetContext = mailetContext;
+        this.blobStore = blobStore;
+        this.fileSystem = fileSystem;
+        this.configuration = configuration;
+    }
+
+    @Override
+    public ShareeStage share(BlobId blobId) {
+        return mailAddress -> explanation ->
+            copyBlobToFile(blobId)
+                .subscribeOn(Schedulers.elastic())
+                .publishOn(Schedulers.elastic())
```

Comment:
this line is supposed to be useless (the `subscribeOn` makes code run on the choosen scheduler already)

Reply:
I decided to remove the use of reactor in fine